packages/visual-editor/src/components/DraggableBlock/EditorBlock.tsx

```diff
@@ -73,7 +73,7 @@ export const EditorBlock: React.FC<EditorBlockProps> = ({
     (state) => state.hoveredComponentId === componentId,
   );
   const coordinates = useSelectedInstanceCoordinates({ node });
-  const displayName = node.data.displayName;
+  const displayName = node.data.displayName || rawNode.data.displayName;
```

Comment:
I tested this out locally and it fixes 99% of the problem. The outermost structure component for the pattern still shows "No label specified" when the outer section is wider than it's first child element (for example Section > Container, or Section > Columns)

https://github.com/user-attachments/assets/df7db5c0-72ac-47af-9de6-31f4a5e50e67

Reply:
nice find, was able to fix it by removing `isAssembly` check